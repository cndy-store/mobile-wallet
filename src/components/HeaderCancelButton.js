import React from 'react';
import PropTypes from 'prop-types';
import { Platform } from 'react-native';
import { Icon, Button, Text } from 'native-base';

class HeaderCancelButton extends React.Component {
  static propTypes = {
    onCancel: PropTypes.func.isRequired,
    icon: PropTypes.string,
    text: PropTypes.string
  };

  static defaultProps = {
    icon: 'arrow-back',
    text: 'Cancel'
  };

  getPlatform() {
    return Platform.OS;
  }

  renderIOS() {
    return (
      <Button transparent onPress={this.props.onCancel}>
        <Text>{this.props.text}</Text>
      </Button>
    );
  }

  renderAndroid() {
    return (
      <Button transparent onPress={this.props.onCancel}>

        <Icon name={this.props.icon} />
      </Button>
    );
  }

  render() {
    if (this.getPlatform() === 'ios') {
      return this.renderIOS();
    }
    return this.renderAndroid();
  }
}

export default HeaderCancelButton;
