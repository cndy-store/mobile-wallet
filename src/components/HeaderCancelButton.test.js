import React from 'react';
import { render } from '../__tests__/renderer';
import { Platform } from 'react-native';
import HeaderCancelButton from './HeaderCancelButton';

it('renders correctly for ios', () => {
  HeaderCancelButton.prototype.getPlatform = jest.fn(() => 'ios');

  const onCancel = jest.fn();
  const { toJSON } = render(<HeaderCancelButton onCancel={onCancel} />);
  expect(toJSON()).toMatchSnapshot();
});

it('renders correctly for android', () => {
  HeaderCancelButton.prototype.getPlatform = jest.fn(() => 'android');

  const onCancel = jest.fn();
  const { toJSON } = render(<HeaderCancelButton onCancel={onCancel} />);
  expect(toJSON()).toMatchSnapshot();
});
