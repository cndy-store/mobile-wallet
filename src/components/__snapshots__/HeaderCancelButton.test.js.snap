// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for android 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 0,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#007aff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "lineHeight": 19,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Cancel
  </Text>
</View>
`;

exports[`renders correctly for ios 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 0,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#007aff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "lineHeight": 19,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Cancel
  </Text>
</View>
`;
